# 설정

In [ ]:
# 한글 폰트 
import matplotlib.font_manager as fm
import matplotlib as mpl

!apt-get update -qq
!apt-get install fonts-nanum* -qq

set(sorted([f.name for f in mpl.font_manager.fontManager.ttflist]))
font_list = fm.findSystemFonts(fontpaths = None, fontext = 'ttf')

mpl.rc('font', family='NanumGothicCoding')
mpl.rc('axes', unicode_minus=False)

fm._rebuild()

Selecting previously unselected package fonts-nanum.
(Reading database ... 160837 files and directories currently installed.)
Preparing to unpack .../fonts-nanum_20170925-1_all.deb ...
Unpacking fonts-nanum (20170925-1) ...
Selecting previously unselected package fonts-nanum-eco.
Preparing to unpack .../fonts-nanum-eco_1.000-6_all.deb ...
Unpacking fonts-nanum-eco (1.000-6) ...
Selecting previously unselected package fonts-nanum-extra.
Preparing to unpack .../fonts-nanum-extra_20170925-1_all.deb ...
Unpacking fonts-nanum-extra (20170925-1) ...
Selecting previously unselected package fonts-nanum-coding.
Preparing to unpack .../fonts-nanum-coding_2.5-1_all.deb ...
Unpacking fonts-nanum-coding (2.5-1) ...
Setting up fonts-nanum-extra (20170925-1) ...
Setting up fonts-nanum (20170925-1) ...
Setting up fonts-nanum-coding (2.5-1) ...
Setting up fonts-nanum-eco (1.000-6) ...
Processing triggers for fontconfig (2.12.6-0ubuntu2) ...


In [ ]:
# retina setting
from IPython.display import set_matplotlib_formats
set_matplotlib_formats('retina')

In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import folium
import matplotlib.pyplot as plt

# csv

In [ ]:
h_2021 = pd.read_csv('/content/drive/MyDrive/Project/house_price/아파트 매매 실거래자료 현황2021.csv', encoding='utf-8')
h_2021

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2718: DtypeWarning: Columns (2) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


시군구      번지   본번  부번  ...   층  건축년도       도로명  해제사유발생일
0      경기도 가평군 가평읍 달전리     281  281   0  ...   3  1996    달전로 59      NaN
1      경기도 가평군 가평읍 달전리     281  281   0  ...   5  1996    달전로 59      NaN
2      경기도 가평군 가평읍 달전리   285-1  285   1  ...   2  1993    달전로 64      NaN
3      경기도 가평군 가평읍 달전리   285-1  285   1  ...  11  1993    달전로 64      NaN
4      경기도 가평군 가평읍 대곡리  230-18  230  18  ...   5  2019  굴다리길 7-1      NaN
...                ...     ...  ...  ..  ...  ..   ...       ...      ...
96671  경기도 화성시 향남읍 행정리     475  475   0  ...   2  2009   행정동로 83      NaN
96672  경기도 화성시 향남읍 행정리     475  475   0  ...  10  2009   행정동로 83      NaN
96673  경기도 화성시 향남읍 행정리     475  475   0  ...   7  2009   행정동로 83      NaN
96674  경기도 화성시 향남읍 행정리     475  475   0  ...  10  2009   행정동로 83      NaN
96675  경기도 화성시 향남읍 행정리     475  475   0  ...  13  2009   행정동로 83      NaN

[96676 rows x 13 columns]

In [ ]:
h_2021.shape

(96676, 13)

'야탑동 4','이매동 8','율동','서현동 5','분당동 1','수내동 5','정자동 7','구미동 2','금곡동 3'

In [ ]:
b1 = h_2021[h_2021['시군구'].str.contains('야탑동')]
b2 = h_2021[h_2021['시군구'].str.contains('이매동')]
b3 = h_2021[h_2021['시군구'].str.contains('율동')]
b4 = h_2021[h_2021['시군구'].str.contains('서현동')]
b5 = h_2021[h_2021['시군구'].str.contains('분당동')]
b6 = h_2021[h_2021['시군구'].str.contains('수내동')]
b7 = h_2021[h_2021['시군구'].str.contains('정자동')]
b8 = h_2021[h_2021['시군구'].str.contains('구미동')]
b9 = h_2021[h_2021['시군구'].str.contains('금곡동')]

In [ ]:
b1.insert(1, 'adm_cd', 4)

In [ ]:
b2.insert(1, 'adm_cd', 8)
b4.insert(1, 'adm_cd', 5)
b5.insert(1, 'adm_cd', 1)
b6.insert(1, 'adm_cd', 5)
b7.insert(1, 'adm_cd', 7)
b8.insert(1, 'adm_cd', 2)
b9.insert(1, 'adm_cd', 3)

분당구 구미동 금곡동 야탑동 서현동 수내동 정자동 이매동

In [ ]:
bundang = pd.concat([b1,b2,b4,b5,b6,b7,b8,b9],ignore_index=True ) # ignore_index=True 
bundang

시군구  adm_cd    번지    본번  ...   층  건축년도      도로명  해제사유발생일
0     경기도 성남분당구 야탑동       4   188   188  ...  17  2003  돌마로 906      NaN
1     경기도 성남분당구 야탑동       4   188   188  ...   3  2003  돌마로 906      NaN
2     경기도 성남분당구 야탑동       4   188   188  ...  13  2003  돌마로 906      NaN
3     경기도 성남분당구 야탑동       4   188   188  ...  15  2003  돌마로 906      NaN
4     경기도 성남분당구 야탑동       4   188   188  ...   4  2003  돌마로 906      NaN
...             ...     ...   ...   ...  ...  ..   ...      ...      ...
2383  경기도 수원권선구 금곡동       3  1147  1147  ...  18  2017   매곡로 43      NaN
2384  경기도 수원권선구 금곡동       3  1147  1147  ...  12  2017   매곡로 43      NaN
2385  경기도 수원권선구 금곡동       3  1147  1147  ...  11  2017   매곡로 43      NaN
2386  경기도 수원권선구 금곡동       3  1147  1147  ...  23  2017   매곡로 43      NaN
2387  경기도 수원권선구 금곡동       3  1147  1147  ...  15  2017   매곡로 43      NaN

[2388 rows x 14 columns]

In [ ]:
mask1 = bundang['전용면적(㎡)'] >= 70
df_80 = bundang.loc[mask1,:]

In [ ]:
df_80

시군구  adm_cd    번지    본번  ...   층  건축년도      도로명  해제사유발생일
0     경기도 성남분당구 야탑동       4   188   188  ...  17  2003  돌마로 906      NaN
1     경기도 성남분당구 야탑동       4   188   188  ...   3  2003  돌마로 906      NaN
2     경기도 성남분당구 야탑동       4   188   188  ...  13  2003  돌마로 906      NaN
3     경기도 성남분당구 야탑동       4   188   188  ...  15  2003  돌마로 906      NaN
4     경기도 성남분당구 야탑동       4   188   188  ...   4  2003  돌마로 906      NaN
...             ...     ...   ...   ...  ...  ..   ...      ...      ...
2383  경기도 수원권선구 금곡동       3  1147  1147  ...  18  2017   매곡로 43      NaN
2384  경기도 수원권선구 금곡동       3  1147  1147  ...  12  2017   매곡로 43      NaN
2385  경기도 수원권선구 금곡동       3  1147  1147  ...  11  2017   매곡로 43      NaN
2386  경기도 수원권선구 금곡동       3  1147  1147  ...  23  2017   매곡로 43      NaN
2387  경기도 수원권선구 금곡동       3  1147  1147  ...  15  2017   매곡로 43      NaN

[2059 rows x 14 columns]

'야탑동 4','이매동 8','율동','서현동 5','분당동 1','수내동 5','정자동 7','구미동 2','금곡동 3'

In [ ]:
c1 = df_80[df_80['시군구'].str.contains('야탑동')]
c2 = df_80[df_80['시군구'].str.contains('이매동')]
c3 = df_80[df_80['시군구'].str.contains('서현동')]
c4 = df_80[df_80['시군구'].str.contains('분당동')]
c5 = df_80[df_80['시군구'].str.contains('수내동')]
c6 = df_80[df_80['시군구'].str.contains('정자동')]
c7 = df_80[df_80['시군구'].str.contains('구미동')]
c8 = df_80[df_80['시군구'].str.contains('금곡동')]

In [ ]:
c1

시군구  adm_cd   번지   본번  부번  ... 거래금액(만원)   층  건축년도      도로명 해제사유발생일
0    경기도 성남분당구 야탑동       4  188  188   0  ...   58,800  17  2003  돌마로 906     NaN
1    경기도 성남분당구 야탑동       4  188  188   0  ...   67,000   3  2003  돌마로 906     NaN
2    경기도 성남분당구 야탑동       4  188  188   0  ...   67,200  13  2003  돌마로 906     NaN
3    경기도 성남분당구 야탑동       4  188  188   0  ...   71,000  15  2003  돌마로 906     NaN
4    경기도 성남분당구 야탑동       4  188  188   0  ...   74,000   4  2003  돌마로 906     NaN
..             ...     ...  ...  ...  ..  ...      ...  ..   ...      ...     ...
168  경기도 성남분당구 야탑동       4  536  536   0  ...   92,000   1  1992   야탑로 20     NaN
169  경기도 성남분당구 야탑동       4  536  536   0  ...  105,000   7  1992   야탑로 20     NaN
170  경기도 성남분당구 야탑동       4  536  536   0  ...  105,000  14  1992   야탑로 20     NaN
171  경기도 성남분당구 야탑동       4  536  536   0  ...  126,000   8  1992   야탑로 20     NaN
172  경기도 성남분당구 야탑동       4  536  536   0  ...  119,500  16  1992   야탑로 20     NaN

[147 rows x 14 columns]



---



In [ ]:
df_bd = bundang.copy()

In [ ]:
import folium
df = pd.read_csv('/content/drive/MyDrive/Project/boondang.csv')
geo_data = '/content/drive/MyDrive/Project/1기 geojson/map (1).geojson'
# 분당 37.369, 127.117 / 일산 37.667, 126.774 / 중동 37.500, 126.764 / 평촌 37.390, 126.960

center = [37.369, 127.117]

m = folium.Map(location=center, zoom_start=13)

# Choropleth 레이어를 만들고, 맵 m에 추가합니다.
folium.Choropleth(
    geo_data=geo_data,
    data=df,
    columns=('adm_cd','평균가격'),
    key_on='feature.properties.adm_cd',
    fill_color='BuPu',
    legend_name='아파트매매 평균가'
).add_to(m)

m

In [ ]:
df

동  adm_cd    평균가격
0  분당동       1   98634
1  구미동       2   95843
2  금곡동       3  107010
3  야탑동       4   91750
4  서현동       5  113462
5  수내동       6  129138
6  정자동       7  115653
7  이매동       8  115573


'야탑동 4', 91750
'이매동 8', 115573
'서현동 5', 113462
'분당동 1', 98634
'수내동 6', 129138
'정자동 7', 115653
'구미동 2', 95843
'금곡동 3', 107010 

In [ ]:
geo_data = '/content/drive/MyDrive/Project/boondang - Sheet1.csv'

# 일산

In [ ]:
import folium
import pandas as pd

df = pd.read_csv('/content/drive/MyDrive/Project/del/ilsan.csv')
geo_data = '/content/drive/MyDrive/Project/del/ilsan.geojson'
# 분당 37.369, 127.117 / 일산 37.667, 126.774 / 중동 37.500, 126.764 / 평촌 37.390, 126.960

center = [37.667, 126.774]

m = folium.Map(location=center, zoom_start=13)

# Choropleth 레이어를 만들고, 맵 m에 추가합니다.
folium.Choropleth(
    geo_data=geo_data,
    data=df,
    columns=('adm_cd','평균가격'),
    key_on='feature.properties.adm_cd',
    fill_color='BuPu',
    legend_name='아파트매매 평균가'
).add_to(m)

m

# 중동


In [ ]:
import folium
import pandas as pd

df = pd.read_csv('/content/drive/MyDrive/Project/del/joongdong.csv')
geo_data = '/content/drive/MyDrive/Project/del/joongdong.geojson'
# 분당 37.369, 127.117 / 일산 37.667, 126.774 / 중동 37.500, 126.764 / 평촌 37.390, 126.960

center = [37.500, 126.764]

m = folium.Map(location=center, zoom_start=14)

# Choropleth 레이어를 만들고, 맵 m에 추가합니다.
folium.Choropleth(
    geo_data=geo_data,
    data=df,
    columns=('adm_cd','평균가격'),
    key_on='feature.properties.adm_cd',
    fill_color='BuPu',
    legend_name='아파트매매 평균가'
).add_to(m)

m

# 평촌

In [ ]:
import folium
df = pd.read_csv('/content/drive/MyDrive/Project/del/pyeongchon.csv')
geo_data = '/content/drive/MyDrive/Project/del/pyeongchon.geojson'
# 분당 37.369, 127.117 / 일산 37.667, 126.774 / 중동 37.500, 126.764 / 평촌 37.390, 126.960

center = [37.390, 126.960]

m = folium.Map(location=center, zoom_start=13)

# Choropleth 레이어를 만들고, 맵 m에 추가합니다.
folium.Choropleth(
    geo_data=geo_data,
    data=df,
    columns=('adm_cd','평균가격'),
    key_on='feature.properties.adm_cd',
    fill_color='BuPu',
    legend_name='아파트매매 평균가'
).add_to(m)

m

# 분당


In [ ]:
import folium
df = pd.read_csv('/content/drive/MyDrive/Project/del/boondang.csv')
geo_data = '/content/drive/MyDrive/Project/del/boondang.geojson'
# 분당 37.369, 127.117 / 일산 37.667, 126.774 / 중동 37.500, 126.764 / 평촌 37.390, 126.960

center = [37.369, 127.117]

m = folium.Map(location=center, zoom_start=13)

# Choropleth 레이어를 만들고, 맵 m에 추가합니다.
folium.Choropleth(
    geo_data=geo_data,
    data=df,
    columns=('adm_cd','평균가격'),
    key_on='feature.properties.adm_cd',
    fill_color='BuPu',
    legend_name='아파트매매 평균가'
).add_to(m)

m

# 1기 신도시

In [ ]:
import folium
df = pd.read_csv('/content/drive/MyDrive/Project/del/new_town_1.csv')
geo_data = '/content/drive/MyDrive/Project/del/new_town_1.geojson'
# 분당 37.369, 127.117 / 일산 37.667, 126.774 / 중동 37.500, 126.764 / 평촌 37.390, 126.960

#중간값 37.523, 921
center = [37.523, 126.921]

m = folium.Map(location=center, zoom_start=11)

# Choropleth 레이어를 만들고, 맵 m에 추가합니다.
folium.Choropleth(
    geo_data=geo_data,
    data=df,
    columns=('adm_cd','평균가격'),
    key_on='feature.properties.adm_cd',
    fill_color='BuPu',
    legend_name='아파트매매 평균가'
).add_to(m)

m

# 위례

In [ ]:
import folium
df = pd.read_csv('/content/drive/MyDrive/Project/del/wirye.csv')
geo_data = '/content/drive/MyDrive/Project/del/wirye.geojson'
# 분당 37.369, 127.117 / 일산 37.667, 126.774 / 중동 37.500, 126.764 / 평촌 37.390, 126.960 #중간값 37.523, 921

# 위례 37.474, 127.151

center = [37.474, 127.151]

m = folium.Map(location=center, zoom_start=14)

# Choropleth 레이어를 만들고, 맵 m에 추가합니다.
folium.Choropleth(
    geo_data=geo_data,
    data=df,
    columns=('adm_cd','평균가격'),
    key_on='feature.properties.adm_cd',
    fill_color='BuPu',
    legend_name='아파트매매 평균가'
).add_to(m)

m

# 판교

In [ ]:
import folium
df = pd.read_csv('/content/drive/MyDrive/Project/del/pankyo.csv')
geo_data = '/content/drive/MyDrive/Project/del/pankyo.geojson'
# 분당 37.369, 127.117 / 일산 37.667, 126.774 / 중동 37.500, 126.764 / 평촌 37.390, 126.960 #중간값 37.523, 921

# 위례 37.474, 127.151 / 판교 37.392, 127.069

center = [37.392, 127.069]

m = folium.Map(location=center, zoom_start=13)

# Choropleth 레이어를 만들고, 맵 m에 추가합니다.
folium.Choropleth(
    geo_data=geo_data,
    data=df,
    columns=('adm_cd','평균가격'),
    key_on='feature.properties.adm_cd',
    fill_color='BuPu',
    legend_name='아파트매매 평균가'
).add_to(m)

m

# 광교

In [ ]:
import folium
df = pd.read_csv('/content/drive/MyDrive/Project/del/kwangkyo.csv')
geo_data = '/content/drive/MyDrive/Project/del/kwangkyo.geojson'
# 분당 37.369, 127.117 / 일산 37.667, 126.774 / 중동 37.500, 126.764 / 평촌 37.390, 126.960 #중간값 37.523, 921

# 위례 37.474, 127.151 / 판교 37.392, 127.069 / 광교 37.291, 127.054

center = [37.291, 127.054]

m = folium.Map(location=center, zoom_start=13)

# Choropleth 레이어를 만들고, 맵 m에 추가합니다.
folium.Choropleth(
    geo_data=geo_data,
    data=df,
    columns=('adm_cd','평균가격'),
    key_on='feature.properties.adm_cd',
    fill_color='BuPu',
    legend_name='아파트매매 평균가'
).add_to(m)

m

# 동탄

In [ ]:
import folium
df = pd.read_csv('/content/drive/MyDrive/Project/del/dongtan.csv')
geo_data = '/content/drive/MyDrive/Project/del/dongtan.geojson'
# 분당 37.369, 127.117 / 일산 37.667, 126.774 / 중동 37.500, 126.764 / 평촌 37.390, 126.960 #중간값 37.523, 921

# 위례 37.474, 127.151 / 판교 37.392, 127.069 / 광교 37.291, 127.054 / 동탄 37.198, 127.084

center = [37.198, 127.084]

m = folium.Map(location=center, zoom_start=13)

# Choropleth 레이어를 만들고, 맵 m에 추가합니다.
folium.Choropleth(
    geo_data=geo_data,
    data=df,
    columns=('adm_cd','평균가격'),
    key_on='feature.properties.adm_cd',
    fill_color='BuPu',
    legend_name='아파트매매 평균가'
).add_to(m)

m

# 2기 신도시

In [ ]:
import folium
import pandas as pd

df = pd.read_csv('/content/drive/MyDrive/Project/del/new_town_2.csv')
geo_data = '/content/drive/MyDrive/Project/del/new_town_2.geojson'
# 분당 37.369, 127.117 / 일산 37.667, 126.774 / 중동 37.500, 126.764 / 평촌 37.390, 126.960 #중간값 37.523, 126.921

# 위례 37.474, 127.151 / 판교 37.392, 127.069 / 광교 37.291, 127.054 / 동탄 37.198, 127.084 #중간값 37.341, 127.099

center = [37.341, 127.099]

m = folium.Map(location=center, zoom_start=11)

# Choropleth 레이어를 만들고, 맵 m에 추가합니다.
folium.Choropleth(
    geo_data=geo_data,
    data=df,
    columns=('adm_cd','평균가격'),
    key_on='feature.properties.adm_cd',
    fill_color='BuPu',
    legend_name='아파트매매 평균가'
).add_to(m)

m

In [ ]:
pip install pydeck

In [ ]:
pip install geopandas

     |████████████████████████████████| 994 kB 4.3 MB/s 
     |████████████████████████████████| 15.4 MB 36 kB/s 
     |████████████████████████████████| 6.6 MB 54.8 MB/s 


In [ ]:
MAPBOX_API_KEY = 'pk.eyJ1IjoicnV0ajk4IiwiYSI6ImNrczQ0ZDlyMTBnNTEyb3B0cXA2aWZydTYifQ.fyNk02fxZZJ-S6kKgJt_QQ'

In [ ]:
import folium
import pydeck as pdk
import pandas as pd
import geopandas as gpd

df = pd.read_csv('/content/drive/MyDrive/Project/del/new_town_2.csv')
geo_data = '/content/drive/MyDrive/Project/del/new_town_2.geojson'

In [ ]:
df

동  adm_cd    평균가격
0           학암동      29  123868
1           창곡동      30  136873
2           운중동      31  131268
3           삼평동      32  141333
4           판교동      33  125775
5           백현동      34  166070
6           이의동      35  113225
7            하동      36   96055
8           원천동      37  101737
9           상현동      38   72318
10  오산동 방교동 금곡동      39   40161
11          반송동      40   59194
12          석우동      41   55535
13           능동      42   47739
14       영천동 중동      43   57992
15          청계동      44   81540
16    신동 산척동 목동      45   69869
17          장지동      46   68126

In [ ]:
# 분당 37.369, 127.117 / 일산 37.667, 126.774 / 중동 37.500, 126.764 / 평촌 37.390, 126.960 #중간값 37.523, 126.921

# 위례 37.474, 127.151 / 판교 37.392, 127.069 / 광교 37.291, 127.054 / 동탄 37.198, 127.084 #중간값 37.341, 127.099

def multipolygon_to_coordinates(x):
    lon, lat = x[0].exterior.xy
    return [[x, y] for x, y in zip(lon, lat)]

df['coordinates'] = df['geometry'].apply(multipolygon_to_coordinates)
del df['geometry']

df[''] = df['인구'] / df['인구'].max()

center = [37.341, 127.099]

m = folium.Map(location=center, zoom_start=11)

# Choropleth 레이어를 만들고, 맵 m에 추가합니다.
folium.Choropleth(
    geo_data=geo_data,
    data=df,
    columns=('adm_cd','평균가격'),
    key_on='feature.properties.adm_cd',
    fill_color='BuPu',
    legend_name='아파트매매 평균가'
).add_to(m)

m

df['정규화인구'] = df['인구'] / df['인구'].max()